packages/contracts-bedrock/src/L2/OptimismSuperchainERC20.sol

```diff
@@ -0,0 +1,187 @@
+    modifier onlyBridge() {
+        if (msg.sender != BRIDGE) revert OnlyBridge();
+        _;
+    }
+
+    /// @notice Semantic version.
+    /// @custom:semver 1.0.0-beta.1
+    string public constant version = "1.0.0-beta.1";
+
+    /// @notice Constructs the OptimismSuperchainERC20 contract.
+    constructor() {
+        _disableInitializers();
+    }
+
+    /// @notice Initializes the contract.
+    /// @param _remoteToken    Address of the corresponding remote token.
+    /// @param _name           ERC20 name.
+    /// @param _symbol         ERC20 symbol.
+    /// @param _decimals       ERC20 decimals.
+    function initialize(
+        address _remoteToken,
+        string memory _name,
+        string memory _symbol,
+        uint8 _decimals
+    )
+        external
+        initializer
+    {
+        OptimismSuperchainERC20Metadata storage _storage = _getMetadataStorage();
+        _storage.remoteToken = _remoteToken;
+        _storage.name = _name;
+        _storage.symbol = _symbol;
+        _storage.decimals = _decimals;
+    }
+
+    /// @notice Allows the L2StandardBridge to mint tokens.
+    /// @param _to     Address to mint tokens to.
+    /// @param _amount Amount of tokens to mint.
+    function mint(address _to, uint256 _amount) external virtual onlyBridge {
+        if (_to == address(0)) revert ZeroAddress();
+
+        _mint(_to, _amount);
+
+        emit Mint(_to, _amount);
+    }
+
+    /// @notice Allows the L2StandardBridge to burn tokens.
+    /// @param _from   Address to burn tokens from.
+    /// @param _amount Amount of tokens to burn.
+    function burn(address _from, uint256 _amount) external virtual onlyBridge {
+        if (_from == address(0)) revert ZeroAddress();
+
+        _burn(_from, _amount);
+
+        emit Burn(_from, _amount);
+    }
+
+    /// @notice Sends tokens to some target address on another chain.
+    /// @param _to      Address to send tokens to.
+    /// @param _amount  Amount of tokens to send.
+    /// @param _chainId Chain ID of the destination chain.
+    function sendERC20(address _to, uint256 _amount, uint256 _chainId) external {
+        if (_to == address(0)) revert ZeroAddress();
+
+        _burn(msg.sender, _amount);
+
+        bytes memory _message = abi.encodeCall(this.relayERC20, (msg.sender, _to, _amount));
+        IL2ToL2CrossDomainMessenger(MESSENGER).sendMessage(_chainId, address(this), _message);
+
+        emit SendERC20(msg.sender, _to, _amount, _chainId);
+    }
+
+    /// @notice Relays tokens received from another chain.
+    /// @param _from   Address of the msg.sender of sendERC20 on the source chain.
+    /// @param _to     Address to relay tokens to.
+    /// @param _amount Amount of tokens to relay.
+    function relayERC20(address _from, address _to, uint256 _amount) external {
+        if (_to == address(0)) revert ZeroAddress();
+
+        if (msg.sender != MESSENGER) revert CallerNotL2ToL2CrossDomainMessenger();
```

Comment:
It is possible that we may not need this check given the check on `crossDomainMessageSender` but feels safer to have it, this is the same pattern as in the `L2ToL2CrossDomainMessenger`

Reply:
Is it possible in any way that an unexpected call can come thru there without this assertion? It would allow for a callback not otherwise expected, but only if an untrusted contract was in the call stack

Reply:
I believe this assertion was thought back when we had the `data` parameter. With the `data` parameter the code called an external contract(`_to`) before the transient data in `L2ToL2CrossDomainMessenger` was set back to 0, which, if the `to` happened to be malicious, would have allowed it to callback `relayERC20` as many times as gas allowed.

With the current code I don't see a way of attacking the logic if this check happened to be gone as the `sendERC20` function doesn't include any other call other than this one in the message. With that being said, it does feel safer to have it.

Reply:
I am on board with keeping it if it helps us sleep better at night